clippy_lints/src/matches/mod.rs

```diff
@@ -1059,7 +1061,28 @@ impl<'tcx> LateLintPass<'tcx> for Matches {
                     }
 
                     redundant_pattern_match::check_match(cx, expr, ex, arms);
-                    single_match::check(cx, ex, arms, expr);
+                    let source_map = cx.tcx.sess.source_map();
+                    let mut match_comments = span_extract_comments(source_map, expr.span);
```

Comment:
Wouldn't it be better to extract the comment spans instead of the text to compare?

What if we have the same comment text twice?

Reply:
It's a `Vec`, not a map. So if a comment is present more than once, not an issue since it'll still remain one at the end. Gonna add a test for this case though.

As for using spans, it's possible but would likely make the code much less readable as we manipulate `InnerSpan`s here since we parse a snippet and not use AST comments.